### ISSUE TITLE
Woocommerce Currency Switcher, Pay for Payment a České služby

### ISSUE THREAD

**Issue by novetrendy** [pavelevap/ceske-sluzby#26]
Při použití více měn s pluginem Woocommerce Currency Switcher je problém, že plugin použitý v Českých službách Pay for Payment přidané poplatky nezahrnuje. Což ve výsledku znamená, že pokud mám například zvolenou měnu Euro - produkty OK, doprava OK, ale polatek za dobírku se nepřepočítá, jen se změní z Kč na Euro. A poplatek za dobírku 30 Euro je celkem hodně.

Níže je dočasné řešení:
V souboru woocommerce-payforpayment.php řádek 190 - 195 nahradit kódem
```
$this->_fee = (object) array(
							'fee_title' => $fee_title,
							'cost' => $cost = apply_filters('woocs_exchange_value', $cost),
							'taxable' => $taxable,
							'tax_class' => $tax_class,
						);
```    
Důležitý je řádek  

```
'cost' => $cost = apply_filters('woocs_exchange_value', $cost),

```
kde jsem použil filtr ```  woocs_exchange_value ```  

Proto prosím o integrci přímo do českých služeb, hlavně kvůli následným aktualizacím pluginu Pay4Payment.

**Comment by pavelevap** [comment]
Nebude to způsobovat problémy pro uživatele, kteří tento plugin nepoužívají? Budu to muset nejdříve vyzkoušet, tento filtr by se měl aplikovat patrně pokud je aktivní...

http://currency-switcher.com/documentation/#!/section_8

Díky za report!

**Comment by novetrendy** [comment]
Určitě to bude muset být přes podmínku, pokud je plugin aktivovaný. 
Já jsem to potřeboval vyřešit velice rychle, proto jsem přepsal přímo plugin, což se nedělá.
Lepší by bylo, kdyby to bylo přímo v českých službách, aby posílal do pay4payment již hodnotu upravenou filtrem v případě aktivace Currency Switcher.

**Comment by pavelevap** [comment]
Asi jsem zcela neodhalil příčinu, proč to u dopravy funguje, zatímco u dobírky nikoli (ještě se na to podívám), ale následující řešení by snad mohlo fungovat: https://github.com/pavelevap/ceske-sluzby/commit/523f8455e7e2e1518450f1db65118cd71ad94a83

@novetrendy: Můžete vyzkoušet? Díky!

**Comment by novetrendy** [comment]
Vyzkoušel jsem Vaši úpravu na vlastní dopravu a funguje. Díky!

**Comment by novetrendy** [comment]
Tak nefunguje, ale chyba bude asi na mé straně viz  kód  #58

**Comment by pavelevap** [comment]
@novetrendy: Takže je tam problém pouze u vlastních úprav? Uloženka a DPD ParcelShop fungují bez problémů? Woocommerce Currency Switcher by měl podporovat defaultní způsoby dopravy, ale pokud přidáváte vlastní, tak musíte doplnit filtr.

**Comment by novetrendy** [comment]
Nefunguje ani u DPD ParcelShop. Souvisí to nějak s nastavením calc_tax ve třídě dopravy.

Pokud je 'calc_tax' => 'per_item', tak spočítá dopravu správně, ale bez daně.
Pokud je 'calc_tax' => 'per_order', anebo není zadané, tak spočítá dopravu správně, ale ukrutnou daň.

**Comment by novetrendy** [comment]
Chyba je někde ve Woocommerce Currency Switcher viz: https://wordpress.org/support/topic/wrong-calculation-shipping-price
Když místo něj použiji Aelia Currency Switcher, tak je doprava ok.

**Comment by pavelevap** [comment]
@novetrendy: Mohl bych poprosit o přístupy? Zkoušel jsem to (cca dva týdny zpět) i s daní a vše se počítalo naprosto správně...

**Comment by novetrendy** [comment]
Pošlu zítra, teď se v tom ještě vrtám.

**Comment by novetrendy** [comment]
Úprava pro Aelia Currency Switcher zatím bez podmínky aktivace pluginu - pro Pay4Payment.
Soubor ceske-sluzby.php od řádku 435
```
function ceske_sluzby_dpd_parcelshop_dobirka_pay4pay( $amount ) {
  $available_shipping = WC()->shipping->load_shipping_methods();
  $chosen_shipping_method = WC()->session->get( 'chosen_shipping_methods' );
  if ( $chosen_shipping_method[0] == "ceske_sluzby_dpd_parcelshop" ) {
    $settings = $available_shipping[ $chosen_shipping_method[0] ]->settings;
    $zeme = WC()->customer->get_shipping_country();
    if ( $zeme == "CZ" ) {
      if ( ! empty( $settings['dpd_parcelshop_dobirka'] ) ) {
        $amount = $settings['dpd_parcelshop_dobirka'];
      }
    }
    if ( $zeme == "SK" ) {
      if ( ! empty( $settings['dpd_parcelshop_dobirka-slovensko'] ) ) {
        $amount = $settings['dpd_parcelshop_dobirka-slovensko'];
      }
    }
    //if ( class_exists( 'WOOCS' ) ) {
    //  $amount = apply_filters( 'woocs_exchange_value', $amount );
    // }
$to_currency = get_woocommerce_currency();
    $from_currency = get_option('woocommerce_currency');
    $amount = apply_filters('wc_aelia_cs_convert', $amount, $from_currency, $to_currency);
  }
  return $amount;
}
```
Přidaný kód
    $to_currency = get_woocommerce_currency();
    $from_currency = get_option('woocommerce_currency');
    $amount = apply_filters('wc_aelia_cs_convert', $amount, $from_currency, $to_currency);

**Comment by novetrendy** [comment]
Hledal jsem všude, ale opravdu je chyba ve Woocommerce Currency Switcher, kde si bere VAT(daně) u dopravy nevím odkud.
Když jsem jednou deaktivoval všechny pluginy a aktivoval výchozí šablonu, tak se daň spočítala OK. Po postupném aktivování pluginů se znovu objevila chyba s počítáním DPH u dopravy. 
Pokud si dobře  pamatuji bylo to Woocommerce EU VAT Assistant.
Poté jsem plugin deaktivoval, ale vše bylo při starém. 
Stejně jako deaktivování všech ostatních pluginů, nastavení výchozí šablony, deaktivování WooCommerce, Currency Switcher atd.

**Comment by pavelevap** [comment]
@novetrendy: Máte pravdu, včera jsem si s tím trochu hrál a problém je s DPH u dopravy - když se změní měna z CZK na EUR, tak DPH zůstane nepřepočítaná v CZK a objeví se tam nakonec přemrštěná cena v EUR. Poslal jsem autorovi popis problému a navrženou opravu, tak to snad projde :-) Potom to funguje bez problému i s poplatkem za dobírku...

Aelia Currency Switcher je pouze [placený plugin](https://aelia.co/shop/currency-switcher-woocommerce/)? Bohužel ho nebudu moci vyzkoušet a integrovat, protože k němu nemám přístup...

**Comment by novetrendy** [comment]
Ano, Aelia je pouze placený plugin. Zatím to řeším tak, že na dopravu dávám DPH jen 2,1%, místo 21%. Někde a nevím kde to DPH zdesetinásobí. Chybu jsem zatím nenašel. Jestli je ve Woo, nebo v pluginech.

**Comment by pavelevap** [comment]
@novetrendy: Výpočet dopravy už by měl v nové verzi 1.1.6 fungovat, tak nezapomeňte po aktualizaci navýšit sazbu DPH :-) Byla to chyba pluginu, donutil jsem to autora opravit :-)

https://wordpress.org/support/topic/vat-values-are-not-switched-to-another-currency-for-shipping

České služby fungují a plugin WooCommerce Currency Switched bez problému podporují. Mohu doplnit i podporu pro Aelia Currency Switcher, ale musel by mi ho dát někdo k dispozici, abych to mohl vyzkoušet.

**Comment by pavelevap** [comment]
Nezkoušel jste někdo, zda už to funguje?

**Comment by pavelevap** [comment]
Budeme to brát za vyřešené, původní plugin byl opraven a plugin České služby ho podporuje.

**Comment by pavelevap** [comment]
Tak ještě detail, zajímavý způsob jak integrovat i další multijazyčné pluginy: https://plugins.trac.wordpress.org/changeset/1494590/flexible-shipping/trunk

**Comment by novetrendy** [comment]
Luxusní :+1:
